### DISCUSSION TITLE
2.0.0-beta1

### DISCUSSION THREAD

**Discussion by cweagans** [cweagans/composer-patches#519] [General]
## What's Changed

A lot of work went into this release. Below are all of the PRs that went into this release, but it might be more straightforward to read the [new documentation](https://docs.cweagans.net/composer-patches/). The 2.0.0 release will definitely work a little differently than you're used to if you've been using the 1.x releases.

* Readme: fix code highlighting by @TomasVotruba in https://github.com/cweagans/composer-patches/pull/87
* Only show debugging info when using the verbose option by @pfrenssen in https://github.com/cweagans/composer-patches/pull/90
* Use system tmp dir - fixes #89 by @erik-seifert in https://github.com/cweagans/composer-patches/pull/96
* Document composer.json patching alternatives by @iamEAP in https://github.com/cweagans/composer-patches/pull/97
* Introduce "composer-exit-on-patch-failure" config option. by @Leksat in https://github.com/cweagans/composer-patches/pull/46
* Fixes #38: Multiple packages using the same patch throws an array to string exception. by @grasmash in https://github.com/cweagans/composer-patches/pull/92
* Renamed variable, which is accidentally modified while ignoring patches by @DevinCarlson in https://github.com/cweagans/composer-patches/pull/103
* Fix short array syntax in Patches.php:157 by @echernyavskiy in https://github.com/cweagans/composer-patches/pull/94
* Remove GIT_DIR by @david-garcia-garcia in https://github.com/cweagans/composer-patches/pull/104
* Use --git-dir instead of GIT_DIR env variable by @daften in https://github.com/cweagans/composer-patches/pull/108
* Fix duplicate entries after failed patch. by @LionsAd in https://github.com/cweagans/composer-patches/pull/143
* Add switch to $patch_levels for Magento 2 Official Patches by @navarr in https://github.com/cweagans/composer-patches/pull/136
* Don't return too early before gathering patches of dependencies by @QuingKhaos in https://github.com/cweagans/composer-patches/pull/151
* Increase post install/update handler priority to support other plugins by @mmenozzi in https://github.com/cweagans/composer-patches/pull/153
* Revamp testing framework by @cweagans in https://github.com/cweagans/composer-patches/pull/162
* Exit on patch failure should use composer extra, not localPackage by @kdebisschop in https://github.com/cweagans/composer-patches/pull/166
* Fixes #148: Incompatible with Git 2.14+. by @danepowell in https://github.com/cweagans/composer-patches/pull/165
* Use cweagans/composer-configurable-plugin by @cweagans in https://github.com/cweagans/composer-patches/pull/164
* Add a patch value object + support a more verbose patch format by @cweagans in https://github.com/cweagans/composer-patches/pull/211
* Split out patch resolution into Composer Capabilities by @cweagans in https://github.com/cweagans/composer-patches/pull/212
* Remove patch report generation by @cweagans in https://github.com/cweagans/composer-patches/pull/213
* Make sure that Patch and PatchCollection objects can be correctly serialized/deserialized by @cweagans in https://github.com/cweagans/composer-patches/pull/214
* Add missing ext-json in composer.json. by @drupol in https://github.com/cweagans/composer-patches/pull/248
* Use strict comparison. by @drupol in https://github.com/cweagans/composer-patches/pull/249
* Add contributing section to readme. by @ElijahLynn in https://github.com/cweagans/composer-patches/pull/264
* Use empty() instead of isset() to check for empty arrays in RootComposer (test coverage 100%) by @ElijahLynn in https://github.com/cweagans/composer-patches/pull/265
* Remove dependency patch resolution by @cweagans in https://github.com/cweagans/composer-patches/pull/267
* [docs] Fix grumphp command in readme. by @ElijahLynn in https://github.com/cweagans/composer-patches/pull/269
* Retry failed downloads once by @Berdir in https://github.com/cweagans/composer-patches/pull/282
* Use drupal/core instead of drupal/drupal in examples by @geerlingguy in https://github.com/cweagans/composer-patches/pull/304
* Add support for Composer version 2 by @Ayesh in https://github.com/cweagans/composer-patches/pull/309
* Add PHP 7.4 to Travis build matrix by @kdambekalns in https://github.com/cweagans/composer-patches/pull/328
* Fix Travis config linter warnings by @kdambekalns in https://github.com/cweagans/composer-patches/pull/327
* Fixes #359: COMPOSER_EXIT_ON_PATCH_FAILURE doesn't do anything by @grasmash in https://github.com/cweagans/composer-patches/pull/360
* Add support for Promises being returned by uninstall in Composer 2.1 by @Seldaek in https://github.com/cweagans/composer-patches/pull/364
* Add note about installing `patch` by @afilina in https://github.com/cweagans/composer-patches/pull/379
* Fix for outdated patch version on FreeBSD by @pesc in https://github.com/cweagans/composer-patches/pull/334
* Add a method to tell if a package has been patched by @phenaproxima in https://github.com/cweagans/composer-patches/pull/392
* Add automated tests, require PHP 8 by @danepowell in https://github.com/cweagans/composer-patches/pull/428
* Fix automated tests by @danepowell in https://github.com/cweagans/composer-patches/pull/429
* Add installedPatches property to fix PHP 8.2 deprecation by @shyim in https://github.com/cweagans/composer-patches/pull/409
* Fix #408: Support proxies by @skyred in https://github.com/cweagans/composer-patches/pull/440
* Bump codeception/module-cli from 2.0.0 to 2.0.1 by @dependabot in https://github.com/cweagans/composer-patches/pull/438
* Automatically merge dependency updates by @danepowell in https://github.com/cweagans/composer-patches/pull/442
* Bump composer/composer from 2.5.0 to 2.5.1 by @dependabot in https://github.com/cweagans/composer-patches/pull/430
* Bump phpro/grumphp from 1.14.0 to 1.15.0 by @dependabot in https://github.com/cweagans/composer-patches/pull/431
* Issue #433 - Codeception configuration by @Sweetchuck in https://github.com/cweagans/composer-patches/pull/434
* Increase test coverage for Resolvers\PatchesFile by @fredden in https://github.com/cweagans/composer-patches/pull/381
* Bump composer/composer from 2.5.1 to 2.5.2 by @dependabot in https://github.com/cweagans/composer-patches/pull/448
* How to store links to issues in patch description by @MurzNN in https://github.com/cweagans/composer-patches/pull/419
* 2.x WIP (don't merge yet) by @cweagans in https://github.com/cweagans/composer-patches/pull/447
* Add funding.yml by @cweagans in https://github.com/cweagans/composer-patches/pull/449
* Add test coverage for command provider by @cweagans in https://github.com/cweagans/composer-patches/pull/450
* Add project-specific depth settings by @cweagans in https://github.com/cweagans/composer-patches/pull/452
* Run all patchers in verbose mode by @cweagans in https://github.com/cweagans/composer-patches/pull/454
* Add --dry-run to patchers by @cweagans in https://github.com/cweagans/composer-patches/pull/455
* GitHub configuration updates by @cweagans in https://github.com/cweagans/composer-patches/pull/456
* Add docs feedback template by @cweagans in https://github.com/cweagans/composer-patches/pull/457
* Tweak template by @cweagans in https://github.com/cweagans/composer-patches/pull/458
* Placeholder docs by @cweagans in https://github.com/cweagans/composer-patches/pull/459
* Add some docs by @cweagans in https://github.com/cweagans/composer-patches/pull/460
* Docs 3 by @cweagans in https://github.com/cweagans/composer-patches/pull/461
* DRY up debug output by @cweagans in https://github.com/cweagans/composer-patches/pull/462
* Add doctor command by @cweagans in https://github.com/cweagans/composer-patches/pull/463
* fix status output by @cweagans in https://github.com/cweagans/composer-patches/pull/464
* Docs 4 by @cweagans in https://github.com/cweagans/composer-patches/pull/465
* more docs by @cweagans in https://github.com/cweagans/composer-patches/pull/466
* fix title by @cweagans in https://github.com/cweagans/composer-patches/pull/467
* Upgrade composer configurable plugin by @cweagans in https://github.com/cweagans/composer-patches/pull/468
* Add composer and io to all events + emit event before throwing exception when patch does not apply by @cweagans in https://github.com/cweagans/composer-patches/pull/470
* Bump composer/composer from 2.5.2 to 2.5.3 by @dependabot in https://github.com/cweagans/composer-patches/pull/473
* Couple of small cleanups by @cweagans in https://github.com/cweagans/composer-patches/pull/476
* Only have git patchers and freeform patcher? by @cweagans in https://github.com/cweagans/composer-patches/pull/472
* Add new patcher: git 'init' patcher by @cweagans in https://github.com/cweagans/composer-patches/pull/471
* Fix typo on docs homepage by @anotherjames in https://github.com/cweagans/composer-patches/pull/481
* Bump composer/composer from 2.5.3 to 2.5.4 by @dependabot in https://github.com/cweagans/composer-patches/pull/483
* Issue #486 - Fix syntax errors and indentations in JSON examples by @Sweetchuck in https://github.com/cweagans/composer-patches/pull/487
* Bump squizlabs/php_codesniffer from 3.7.1 to 3.7.2 by @dependabot in https://github.com/cweagans/composer-patches/pull/488
* Rename patches.lock to patches.lock.json by @cweagans in https://github.com/cweagans/composer-patches/pull/492
* Misc updates by @cweagans in https://github.com/cweagans/composer-patches/pull/494
* Bump actions/checkout from 2 to 3 by @dependabot in https://github.com/cweagans/composer-patches/pull/497
* Deduplicate patch collection as patches are added by @cweagans in https://github.com/cweagans/composer-patches/pull/496
* Handle $COMPOSER environment variable by @cweagans in https://github.com/cweagans/composer-patches/pull/493
* Bump composer/composer from 2.5.4 to 2.5.5 by @dependabot in https://github.com/cweagans/composer-patches/pull/501
* Bump guzzlehttp/psr7 from 2.4.3 to 2.5.0 by @dependabot in https://github.com/cweagans/composer-patches/pull/508
* Fix typo by @pepijnolivier in https://github.com/cweagans/composer-patches/pull/504
* Bump phpro/grumphp from 1.15.0 to 1.16.0 by @dependabot in https://github.com/cweagans/composer-patches/pull/509
* Bump composer/composer from 2.5.5 to 2.5.7 by @dependabot in https://github.com/cweagans/composer-patches/pull/513
* Bump composer/composer from 2.5.7 to 2.5.8 by @dependabot in https://github.com/cweagans/composer-patches/pull/515

## New Contributors
* @TomasVotruba made their first contribution in https://github.com/cweagans/composer-patches/pull/87
* @erik-seifert made their first contribution in https://github.com/cweagans/composer-patches/pull/96
* @iamEAP made their first contribution in https://github.com/cweagans/composer-patches/pull/97
* @Leksat made their first contribution in https://github.com/cweagans/composer-patches/pull/46
* @grasmash made their first contribution in https://github.com/cweagans/composer-patches/pull/92
* @DevinCarlson made their first contribution in https://github.com/cweagans/composer-patches/pull/103
* @echernyavskiy made their first contribution in https://github.com/cweagans/composer-patches/pull/94
* @david-garcia-garcia made their first contribution in https://github.com/cweagans/composer-patches/pull/104
* @daften made their first contribution in https://github.com/cweagans/composer-patches/pull/108
* @navarr made their first contribution in https://github.com/cweagans/composer-patches/pull/136
* @QuingKhaos made their first contribution in https://github.com/cweagans/composer-patches/pull/151
* @mmenozzi made their first contribution in https://github.com/cweagans/composer-patches/pull/153
* @cweagans made their first contribution in https://github.com/cweagans/composer-patches/pull/162
* @kdebisschop made their first contribution in https://github.com/cweagans/composer-patches/pull/166
* @drupol made their first contribution in https://github.com/cweagans/composer-patches/pull/248
* @ElijahLynn made their first contribution in https://github.com/cweagans/composer-patches/pull/264
* @geerlingguy made their first contribution in https://github.com/cweagans/composer-patches/pull/304
* @Ayesh made their first contribution in https://github.com/cweagans/composer-patches/pull/309
* @kdambekalns made their first contribution in https://github.com/cweagans/composer-patches/pull/328
* @afilina made their first contribution in https://github.com/cweagans/composer-patches/pull/379
* @pesc made their first contribution in https://github.com/cweagans/composer-patches/pull/334
* @shyim made their first contribution in https://github.com/cweagans/composer-patches/pull/409
* @skyred made their first contribution in https://github.com/cweagans/composer-patches/pull/440
* @dependabot made their first contribution in https://github.com/cweagans/composer-patches/pull/438
* @Sweetchuck made their first contribution in https://github.com/cweagans/composer-patches/pull/434
* @MurzNN made their first contribution in https://github.com/cweagans/composer-patches/pull/419
* @anotherjames made their first contribution in https://github.com/cweagans/composer-patches/pull/481
* @pepijnolivier made their first contribution in https://github.com/cweagans/composer-patches/pull/504

**Full Changelog**: https://github.com/cweagans/composer-patches/compare/1.6.0...2.0.0-beta1

<hr /><em>This discussion was created from the release <a href='https://github.com/cweagans/composer-patches/releases/tag/2.0.0-beta1'>2.0.0-beta1</a>.</em>

**Comment by TomasVotruba** [comment]
Awesome work, thanks for the next release 🤙🎉😊

Look forward to the stable so our package can build on it

**Reply by cweagans** [reply]
Are you extending Composer Patches? If so, now is a great time to build. The api shouldn't change much, but if you run into problems, things can still be tweaked.

**Reply by TomasVotruba** [reply]
Not really, but using the mechanism.

I prefer to update the dependency, fix BC changes and do the release of new version same day :)

**Reply by cweagans** [reply]
Cool! If you're able, I'd appreciate testing with the beta. I can write all the automated tests I want but at the end of the day, the only thing that actually matters is that it works in real projects. :)

**Comment by gitressa** [comment]
Thank you for your great work with Composer Patches, it's an awesome tool. And wow, that's an impressive list of tasks, which have been completed!

I am sorry to hear you got (and maybe still is) burned out. One way to prevent this is keeping things lean, as you already do, and reject complex features, for now.

Maybe you can reach out to some of the top contributors, and invite them to become co-maintainers, to lift some weight off your shoulders?
